agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -408,31 +412,16 @@ impl DeriveSchema for Stage {
             set_windows: &SetWindowFields,
             state: &mut ResultSetState,
         ) -> Result<Schema> {
-            let mut keys = set_windows
-                .output
-                .iter()
-                .map(|(k, e)| {
-                    let field_schema = e.window_func.derive_schema(state)?;
-                    Ok((k.to_string(), field_schema))
-                })
-                .collect::<Result<BTreeMap<_, _>>>()?;
-            let Schema::Document(Document {
-                keys: ref current_keys,
-                ..
-            }) = state.result_set_schema
-            else {
-                // This should never actually happen
-                panic!(
-                    "Found ResultSetSchema that is not a Document: {:?}",
-                    state.result_set_schema
-                );
-            };
-            keys.extend(current_keys.clone());
-            Ok(Schema::Document(Document {
-                required: keys.keys().cloned().collect(),
-                keys,
-                ..Default::default()
-            }))
+            let mut result_doc = state.result_set_schema.clone();
+            for (k, e) in set_windows.output.clone() {
+                let field_schema = e
+                    .window_func
+                    .derive_schema(state)?
+                    .upconvert_missing_to_null();
+                let path = k.split('.').map(|s| s.to_string()).collect::<Vec<String>>();
```

Comment:
This change just a. upconverts missing to null and b. ensures field name is being treated as a path, not a literal